ipaplatform/redhat/authconfig.py

```diff
@@ -18,15 +18,102 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import logging
+
 from ipaplatform.paths import paths
 from ipapython import ipautil
 from ipapython.admintool import ScriptError
 import os
 
 FILES_TO_NOT_BACKUP = ['passwd', 'group', 'shadow', 'gshadow']
 
+logger = logging.getLogger(__name__)
+
+
+class RedHatAuthTool(object):
+    auth_tool = None
+
+    @classmethod
+    def getInstance(cls):
+        if cls.auth_tool is None:
+            cls.auth_tool = RedHatAuthSelect()
+        return cls.auth_tool
+
+
+class RedHatAuthToolBase(object):
```

Comment:
It would be easier and more readable to turn the base class into an abstract base class. The ABC system ensures that it is not possible to create instances of classes that don't implement all abstract methods.

```
@six.add_metaclass(abc.ABCMeta)
class RedHatAuthToolBase(object):

    @abc.abstractmethod
    def configure(self, sssd, mkhomedir, statestore):
        pass
    ...
```

Reply:
I am not sure that we will keep this base class as well. During the feature discussion we kind of wanted to have old behavior be available as well. So, I created is as a kind of interface to the actual tool.
Anyway, will do suggested changes